src/yield/library.cairo

```diff
@@ -260,110 +362,41 @@ namespace CarbonableYielder {
 
-        // [Check] Yielder contribution overflow
-        with_attr error_message("CarbonableYielder: user yielder contribution overflow detected") {
-            assert_in_range(previous_yielder_absorption, 0, current_yielder_absorption + 1);
+        // [Check] User contribution overflow
+        with_attr error_message("CarbonableYielder: user contribution overflow detected") {
+            assert_in_range(previous_user_absorption, 0, current_user_absorption + 1);
         }
-        let period_contribution = current_yielder_absorption - previous_yielder_absorption;
+        let period_contribution = current_user_absorption - previous_user_absorption;
 
         // [Effect] Store new snapshoted absorptions and contibution
-        CarbonableYielder_snapshoted_user_yielder_absorption_.write(
-            user, current_yielder_absorption
-        );
+        CarbonableYielder_snapshoted_user_yielder_absorption_.write(user, current_user_absorption);
         CarbonableYielder_snapshoted_user_yielder_contribution_.write(user, period_contribution);
 
-        // [Check] If not last, then continue
-        if (index != 0) {
-            _snapshot_iter(index=index - 1);
-            return ();
-        }
-        return ();
-    }
-
-    func _create_vestings_iter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        contract_address: felt,
-        yielder_contribution: felt,
-        total_amount: felt,
-        cliff_delta: felt,
-        start: felt,
-        duration: felt,
-        slice_period_seconds: felt,
-        revocable: felt,
-        carbonable_project_address,
-        current_time,
-        index: felt,
-    ) {
-        alloc_locals;
-
-        let (beneficiary) = CarbonableOffseter.user_by_index(index);
-        let (user_contribution) = CarbonableYielder_snapshoted_user_yielder_contribution_.read(
-            beneficiary
-        );
-
-        // [Check] If user abosrption is null, then continue
-        if (user_contribution == 0) {
-            // [Check] If index is null, then stop
-            if (index == 0) {
-                return ();
-            }
-
-            return _create_vestings_iter(
-                contract_address=contract_address,
-                yielder_contribution=yielder_contribution,
-                total_amount=total_amount,
-                cliff_delta=cliff_delta,
-                start=start,
-                duration=duration,
-                slice_period_seconds=slice_period_seconds,
-                revocable=revocable,
-                carbonable_project_address=carbonable_project_address,
-                current_time=current_time,
-                index=index - 1,
-            );
+        // [Effect] Update user claimable if total != 0
+        if (total != 0) {
+            let (stored_claimable) = CarbonableYielder_claimable_.read(user);
+            let (new_claimable, _) = unsigned_div_rem(previous_user_contribution * amount, total);
```

Comment:
Consider to store the sum of reminders and add it to the next amount to distribute